### PR TITLE
Add compatibility with IE versions <= 10

### PR DESCRIPTION
**Pull request by jaredklewis**
This PR sets the `XmlHttpRequest` object's `withCredentials` property AFTER calling the `open` method, instead of before.

Doing so allows this library to be used with IE versions <= 10. Previous to IE11, Internet Explorer would throw an `InvalidStateError` if the `withCredentials` property was set before calling the open method.

Setting `withCredentials` after the `open` call is equally standards compliant.

> 1. If state is not unsent or opened, throw an InvalidStateError exception. 
> 2. If the send() flag is set, throw an InvalidStateError exception. 
> 
> https://xhr.spec.whatwg.org/#the-withcredentials-attribute

In other words, you can set it before opening an XHR, or after opening but before sending. And indeed, in other browsers (Chrome, Firefox, Safari), either method works equally well.

This is a known IE bug, and this PR's fix has been used effectively in other projects, for example: https://github.com/enyo/dropzone/issues/179

### CODE REVIEW

**Comment by skogsmaskin** [comment]
:+1:

**Comment by skogsmaskin** [comment]
@radiosilence could you merge this please? xr is broken for IE in the current version.
